application/controllers/exercises.py

```diff
@@ -0,0 +1,101 @@
+from flask_restful import Resource, reqparse, request
+from flask_restful import fields, marshal_with, marshal
+from application.models.exercise import Exercise
+from application import db
+
+exercise_fields = {
+    'id': fields.Integer,
+    'url': fields.String,
+    'duration': fields.String,
+    'category': fields.String
+}
+
+exercise_list_fields = {
+    'count': fields.Integer,
+    'exercises': fields.List(fields.Nested(exercise_fields)),
+}
+
+exercise_post_parser = reqparse.RequestParser()
+exercise_post_parser.add_argument(
+    'url',
+    type=str,
+    required=True,
+    location=['json'],
+    help='url parameter is required'
+)
+exercise_post_parser.add_argument(
+    'duration',
+    type=str,
+    required=True,
+    location=['json'],
+    help='duration parameter is required'
+)
+exercise_post_parser.add_argument(
+    'category',
+    type=str,
+    required=True,
+    location=['json'],
+    help='category parameter is required'
+)
+
+class ExercisesResource(Resource):
+    def get(self, exercise_id=None):
+        if exercise_id:
+            exercise = Exercise.query.filter_by(id=exercise_id).first()
+            return marshal(exercise, exercise_fields)
+        else:
+            args = request.args.to_dict()
+            limit = args.get('limit', 0)
+            offset = args.get('offset', 0)
+
+            args.pop('limit', None)
+            args.pop('offset', None)
+
+            exercise = Exercise.query.filter_by(**args).order_by(Exercise.id)
```

Comment:
Looking at this query, I assume this means that we would get back all exercises within a category or duration when we make this request? We had talked about that it would be awesome if there was a way for us to hit an endpoint & just back back 1 random video within the selected category/durations if that's at all workable for you all!